### DISCUSSION TITLE
About openid certificates hostname

### DISCUSSION THREAD

**Discussion by hyzx86** [OrchardCMS/OrchardCore#14070] [General]
If the user changes RequestUrlHosts, the old certificate file will still exist

https://github.com/OrchardCMS/OrchardCore/blob/8cc817c13d12ec39f4dbdec6f9c71195c0c511c1/src/OrchardCore.Modules/OrchardCore.OpenId/Services/OpenIdServerService.cs#L589-L600

And then using the new domain name to decrypt the old certificate will cause problems

https://github.com/OrchardCMS/OrchardCore/blob/8cc817c13d12ec39f4dbdec6f9c71195c0c511c1/src/OrchardCore.Modules/OrchardCore.OpenId/Services/OpenIdServerService.cs#L373-L379

Can we consider limiting the domain name when retrieving certificates, 
or more intuitively change the certificate file name to use the domain name as the prefix? In this way, users can intuitively distinguish those unnecessary certificate files

**Comment by sebastienros** [comment]
cc @kevinchalet (no request, just FYI)

**Comment by kevinchalet** [comment]
> And then using the new domain name to decrypt the old certificate will cause problems

It's not clear to me why, as we don't use the subject for either the lookup or the decryption process (it's mostly just a display name that no one will see, unlike a TLS certificate, whose content is made accessible by the browser).

Anyway, we'll likely stop using certificates completely once https://github.com/OrchardCMS/OrchardCore/pull/7891 is merged and the OpenID module is updated to use it and store raw RSA/ECDSA keys.

**Comment by jtkech** [comment]
I will try soon to complete the secrets module PR ;)

**Comment by kevinchalet** [comment]
@jtkech nice! You should get in touch with @Piedone: IIRC, he was interested in funding this task 😃

**Comment by Piedone** [comment]
Yep!

**Comment by jtkech** [comment]
Already done!

**Comment by hyzx86** [comment]
@kevinchalet  Thanks ，You're right, maybe not the problem, because I had a problem the other day, what is the condition to decrypt, machine name or domain name.

https://github.com/OrchardCMS/OrchardCore/discussions/13945

Because I couldn't find the reason, I had to export everything through the recipe and then import it into the new tenant, which is working fine so far
